src/modules/myData/CheckSMSRequestSuccess.tsx

```diff
@@ -0,0 +1,56 @@
+import { Address } from '@/types';
+import { useQuery } from '@tanstack/react-query';
+import { AlertCircle } from 'react-feather';
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from '@/components/ui/tooltip';
+import { getDataProtectorCoreClient } from '@/externals/iexecSdkClient';
+
+export default function CheckSMSRequestSuccess({
+  protectedDataAddress,
+  className,
+}: {
+  className?: string;
+  protectedDataAddress: Address | undefined;
+}) {
+  const {
+    data: hasSmsSecret,
+    isSuccess: isSmsRequestSuccess,
+    isError: isCheckSmsSecretError,
+  } = useQuery({
+    queryKey: ['oneProtectedDataSmsSecret', protectedDataAddress],
+    queryFn: async () => {
+      const dataProtectorCore = await getDataProtectorCoreClient();
+      // @ts-expect-error 'iexec' is a protected field but that's fine
+      return dataProtectorCore.iexec.dataset.checkDatasetSecretExists(
+        protectedDataAddress!
```

Comment:
```suggestion
        protectedDataAddress
```
i think we can avoid using the non-null assertion operator (!) when type checking is already taken care of by the enabled parameter

Reply:
We need it to avoid a warning